src/tests/profiler/native/nongcheap/nongcheap.cpp

```diff
@@ -0,0 +1,74 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "nongcheap.h"
+
+GUID NonGcHeapProfiler::GetClsid()
+{
+    // {EF0D191C-3FC7-4311-88AF-E474CBEB2859}
+    GUID clsid = { 0xef0d191c, 0x3fc7, 0x4311, { 0x88, 0xaf, 0xe4, 0x74, 0xcb, 0xeb, 0x28, 0x59 } };
+    return clsid;
+}
+
+HRESULT NonGcHeapProfiler::Initialize(IUnknown* pICorProfilerInfoUnk)
+{
+    Profiler::Initialize(pICorProfilerInfoUnk);
+
+    HRESULT hr = S_OK;
+    if (FAILED(hr = pCorProfilerInfo->SetEventMask2(
+        COR_PRF_ENABLE_OBJECT_ALLOCATED | COR_PRF_MONITOR_OBJECT_ALLOCATED,
+        COR_PRF_HIGH_BASIC_GC)))
+    {
+        printf("FAIL: ICorProfilerInfo::SetEventMask2() failed hr=0x%x", hr);
+        return hr;
+    }
+
+    return S_OK;
+}
+
+HRESULT STDMETHODCALLTYPE NonGcHeapProfiler::ObjectAllocated(ObjectID objectId, ClassID classId)
+{
+    COR_PRF_GC_GENERATION_RANGE gen;
+    HRESULT hr = pCorProfilerInfo->GetObjectGeneration(objectId, &gen);
+
+    // non-GC objects (same for GC.GetGeneration() API) have generation = -1
+    if (gen.generation == (COR_PRF_GC_GENERATION)INT32_MAX)
+    {
+        if (!FAILED(hr))
```

Comment:
How about calling [this](https://learn.microsoft.com/en-us/dotnet/framework/unmanaged-api/profiling/icorprofilerinfo10-isfrozenobject-method) API to verify that it is a frozen object as well?

Reply:
From my understanding we sort of ignore these old Frozen* profiler APIs as we re-branded Frozen as NonGC. For the same reason we introduce a new API to traverse frozen objects and ignore existing `EnumModuleFrozenObjects` and introduce a new one.

Reply:
Ignore is probably fine when we were still designing, but it is really lousy from a testing perspective. It should either supposed to work or it doesn't. If it is supposed to work, and we are not testing it, it would be a test hole. If it is not supposed to work, then we should update the code to make it fails. If we wanted to depreciate old APIs in favor of new ones, we should make that clear in the documentation. Either way, something is missing.